src/accordion/types.ts

```diff
@@ -0,0 +1,168 @@
+   * @default true
+   */
+  manual: boolean;
+  /**
+   * Allow to toggle accordion items
+   * @default false
+   */
+  allowToggle: boolean;
+}
+
+export type SelectedIdPair = {
+  /**
+   * The current selected accordion's `id`.
+   */
+  selectedId?: string | null;
+  /**
+   * Initial selected accordion's `id`.
+   * @default []
+   */
+  selectedIds?: (string | null)[];
+  setSelectedId?: React.Dispatch<React.SetStateAction<string | null>>;
+  setSelectedIds?: React.Dispatch<React.SetStateAction<(string | null)[]>>;
+};
+
+export interface SingleAccordionProps extends CommonAccordionProps {
+  /**
+   * Allow to open multiple accordion items
+   * @default false
+   */
+  allowMultiple?: false;
+  /**
+   * The current selected accordion's `id`.
+   */
+  selectedId?: string | null;
+  /**
+   * Set default selected id(uncontrolled)
+   *
+   * @default null
+   */
+  defaultSelectedId?: string | null;
+  /**
+   * Handler that is called when the selectedId changes.
+   */
+  onSelectedIdChange?: (value: string | null) => void;
+}
+
+export interface MultiAccordionProps extends CommonAccordionProps {
+  /**
+   * Allow to open multiple accordion items
+   * @default true
+   */
+  allowMultiple?: true;
+  /**
+   * Initial selected accordion's `id`.
+   * @default []
+   */
+  selectedIds?: (string | null)[];
+  /**
+   * Set default selected ids(uncontrolled)
+   *
+   * @default []
+   */
+  defaultSelectedIds?: (string | null)[];
+  /**
+   * Handler that is called when the selectedIds changes.
+   */
+  onSelectedIdsChange?: (value: (string | null)[]) => void;
+}
+
+// State return types //
+
+export interface SingleReturn {
+  /**
+   * Allow to open multiple accordion items
+   * @default false
+   */
+  allowMultiple: false;
+  /**
+   * The current selected accordion's `id`.
+   */
+  selectedId: string | null | undefined;
+  /**
+   * Sets `selectedId`.
+   */
+  setSelectedId: React.Dispatch<React.SetStateAction<string | null>>;
+}
+
+export interface MultiReturn {
+  /**
+   * Allow to open multiple accordion items
+   * @default true
+   */
+  allowMultiple: true;
+  /**
+   * The current selected accordion's `id`.
+   */
+  selectedIds: (string | null)[] | undefined;
+  /**
+   * Sets `selectedIds`.
+   */
+  setSelectedIds: React.Dispatch<React.SetStateAction<(string | null)[]>>;
+}
+
+// Overload signatures for useAccordionState
+export type SingleOverloadReturn = AccordionActions &
+  AccordionState &
+  SingleReturn;
+export type MultiOverloadReturn = AccordionActions &
+  AccordionState &
+  MultiReturn;
+
+type AccordionProps = SingleAccordionProps | MultiAccordionProps;
+export type AccordionReturns = MultiReturn | SingleReturn;
+
+export type AccordionInitialState = CompositeInitialState & AccordionProps;
+export type AccordionInitialStateSingle = CompositeInitialState &
+  SingleAccordionProps;
+export type AccordionInitialStateMulti = CompositeInitialState &
+  MultiAccordionProps;
```

Comment:
What about setting <Partial> here instead of doing it inside the components 🤔

Reply:
Can you be specific? what setting?

Reply:
Oops, that should be setting `<Partial>`.

I did not put them in code blocks 😅

Reply:
Yeah i think we can put in partial here i left it since i was doing experiments and wanted full control